lib/wigets/app_header.dart

```diff
@@ -0,0 +1,200 @@
+		return InkWell(
+			customBorder: new CircleBorder(),
+			onTap: action,
+			child: Padding(
+				padding: EdgeInsets.all(8.0),
+				child:Icon(
+					icon,
+					size: 26.0,
+					color: Colors.white
+				)
+			)
+		);
+	}
+	
+	Widget headerActionButton(BuildContext context, HeaderActionButton button) {
+		return Container(
+			padding: EdgeInsets.all(4.0),
+			child: FlatButton(
+				onPressed: button.action,
+				color: (button.backgroundColor != null) ? button.backgroundColor : Theme.of(context).buttonColor,
+				padding: EdgeInsets.symmetric(vertical: 8.0, horizontal: 10.0),
+				child: (button.customContent != null) ? button.customContent : Row(
+					mainAxisAlignment: MainAxisAlignment.center,
+					children: <Widget>[
+						Padding(
+							child: Icon(button.icon, color: Colors.white, size: 20),
+							padding: EdgeInsets.only(right: AppBoxProperties.buttonIconPadding)
+						),
+						Text(
+							AppLocales.of(context).translate(button.text),
+							style: Theme.of(context).textTheme.button
+						)
+					],
+				)
+			)
+		);
+	}
+
+	Widget headerTextField(BuildContext context, String text) {
+		return Container(
+			alignment: Alignment.centerLeft,
+			padding: EdgeInsets.only(left: 4.0, right: 4.0, top: 8.0, bottom: 8.0),
+			child: Text(
+				AppLocales.of(context).translate(text),
+				textAlign: TextAlign.left,
+				style: Theme.of(context).textTheme.bodyText1
+			)
+		);
+	}
+
+	Widget buildHederContainer(BuildContext context, Widget innerContent) {
+		return Material(
+			elevation: 4.0,
+			color: Theme.of(context).appBarTheme.color,
+			child: Container(
+				padding: EdgeInsets.symmetric(vertical: 10.0, horizontal: 8.0),
+				child: SafeArea(
+					child: Column(
+						children: <Widget>[
+							innerContent,
+							if (text != null)
+								headerTextField(context, text),
+							Container(
+								height: 48,
+								alignment: Alignment.centerLeft,
+								child: ListView(
+									physics: BouncingScrollPhysics(),
+									shrinkWrap: true,
+									scrollDirection: Axis.horizontal,
+									children: headerActionButtons.map((element) => 
+										headerActionButton(context, element)).toList()
+								)
+							)
+						]
+					)
+				)
+			)
+		);
+	}
+
+	Widget buildGreetings(BuildContext context) {
+		return buildHederContainer(context, 
+			Row(
+				mainAxisAlignment: MainAxisAlignment.spaceBetween,
+				crossAxisAlignment: CrossAxisAlignment.start,
+				children: <Widget>[
+					Row(
+						children: <Widget>[
+							Padding(
+								padding: EdgeInsets.only(left: 4.0, right: 8.0),
+								child: headerImage(user)
+							),
+							Column(
+								crossAxisAlignment: CrossAxisAlignment.start,
+								children: <Widget>[
+									RichText(
+										text: TextSpan(
+											text: '${AppLocales.of(context).translate('page.' + ((user.type == UserType.caregiver) ? 'caregiverSection' : 'childSection') + '.panel.header.greetings')},\n',
```

Comment:
That could be broken down as there is an if and a function call all in one string